### PR TITLE
Update semver to >=4.3.2 to resolve #72

### LINKED ISSUE #72: nsp failing its own audit :)
```
Name    Installed  Patched  Vulnerable Dependency
semver    3.0.1    >=4.3.2  page-server@1.0.12 > nsp@1.0.1 > semver@3.0.1
semver    2.3.2    >=4.3.2  page-server@1.0.12 > nsp@1.0.1 > silent-npm-registry-client@0.0.1 > npm-registry-client@0.2.31 > semver@2.3.2
```

### PR DESCRIPTION
**Pull request by chetanddesai**
I generated a new `npm-shrinkwrap.json`.  If you guys want, I can just manually edit it and change only the `semver` dependency.

### CODE REVIEW

**Comment by coreycaverly** [comment]
+1

**Comment by turboki** [comment]
+1

**Comment by chetanddesai** [comment]
Wait, that change only updated `semver`.  Let me add another update that uses the [`1.0.0`](https://github.com/juliangruber/silent-npm-registry-client/commits/master) release of `silent-npm-registry-client` which then will completely resolve #72

File: lib/auditPackage.js [hunk 1]
```diff
@@ -152,30 +154,17 @@ function auditPackage(pkgPath, cb) {
                 return cb2(null);
             }
 
-            registry.get(dependencyName, dependencyVersion,
+            registry.get(registryBaseURL + dependencyName + '/' + dependencyVersion, {timeout: 5000},
```
Comment (chetanddesai) [inline lib/auditPackage.js]: This get API has changed from `npm-registry-client@0.2.31` to `npm-registry-client@6.3.2`.  It now takes a fully qualified URI.  However, it returns the latest version that it resolves, e.g. https://registry.npmjs.org/qs/^0.5 will resolve correctly.  Thus, I was able to rip out a bunch of code that does the `semver.maxSatisfying()` work.  Pretty neat.

**Comment by FGRibreau** [comment]
:+1:

**Comment by bear** [comment]
I'll get this reviewed by the team first thing Monday and work to getting it merged.

Thanks for the update and the work on the project!

**Comment by chetanddesai** [comment]
Sweet, thanks @bear

**Comment by nlf** [comment]
Nice, thanks for this!

**Pull request merged**